src/meshlink.c

```diff
@@ -588,23 +588,24 @@ static bool try_bind(int port) {
 		return false;
 	}
 
-	while(ai) {
-		int fd = socket(ai->ai_family, SOCK_STREAM, IPPROTO_TCP);
+	//while(ai) {
+	for(struct addrinfo *aip = ai; aip; aip = aip->ai_next) {
+		int fd = socket(aip->ai_family, SOCK_STREAM, IPPROTO_TCP);
 
 		if(!fd) {
 			freeaddrinfo(ai);
 			return false;
 		}
 
-		int result = bind(fd, ai->ai_addr, ai->ai_addrlen);
+		int result = bind(fd, aip->ai_addr, aip->ai_addrlen);
 		closesocket(fd);
 
 		if(result) {
 			freeaddrinfo(ai);
 			return false;
 		}
 
-		ai = ai->ai_next;
+		aip = aip->ai_next;
```

Comment:
You're already going to the next pointer in the `for`-statement, so this one should just be removed.

Reply:
I removed that statement